### PR TITLE
Deny clippy warnings in CI properly

### PR DESCRIPTION
**Pull request by Riateche**
Pre-commit config in CI previously used `clippy --fix`, which makes `-D warnings` have no effect. Now warnings should be properly denied. I fixed the simple warnings and added allow annonations for more complex cases.

### CODE REVIEW

File: apps/fortuna/src/api/revelation.rs [hunk 1]
```diff
@@ -40,10 +40,6 @@ pub async fn revelation(
         })
         .inc();
 
-    let sequence: u64 = sequence
-        .try_into()
-        .map_err(|_| RestError::InvalidSequenceNumber)?;
```
Comment (keyvankhademi) [inline apps/fortuna/src/api/revelation.rs:45]: Are you sure the intention here was not to check if `try_into()` has some errors?

File: target_chains/solana/pyth_solana_receiver_sdk/src/cpi/accounts.rs [hunk 2]
```diff
@@ -1,3 +1,5 @@
+#![allow(clippy::vec_init_then_push)]
```
Comment (ali-bahjati) [inline target_chains/solana/pyth_solana_receiver_sdk/src/cpi/accounts.rs:1]: couldn't we move this in?

**Review by ali-bahjati — APPROVED** [review]

File: apps/fortuna/src/api/revelation.rs (hunk 1 above)
Comment (Riateche) [reply apps/fortuna/src/api/revelation.rs:45]: `sequence` is already a `u64` so `try_into()` never fails.

File: target_chains/solana/pyth_solana_receiver_sdk/src/cpi/accounts.rs (hunk 2 above)
Comment (Riateche) [reply target_chains/solana/pyth_solana_receiver_sdk/src/cpi/accounts.rs:1]: The module contains auto generated code so I feel like it's more convenient to keep this outside so it's easier to update the code if needed.

**Pull request merged**